kratos/processes/set_initial_state_process.cpp

```diff
@@ -0,0 +1,210 @@
+
+        KRATOS_CATCH("")
+    }
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+    template<std::size_t TDim>
+    void SetInitialStateProcess<TDim>::ExecuteInitializeSolutionStep()
+    {
+        KRATOS_TRY
+        const auto it_elem_begin = mrModelPart.ElementsBegin();
+        const auto& r_integration_points = it_elem_begin->GetGeometry().
+            IntegrationPoints(it_elem_begin->GetIntegrationMethod());
+
+        Vector aux_initial_strain = mInitialStrain;
+        Vector aux_initial_stress = mInitialStress;
+        Matrix aux_initial_F      = mInitialF;
+        InitialState::Pointer p_initial_state = Kratos::make_intrusive<InitialState>
+            (aux_initial_strain, aux_initial_stress, aux_initial_F);
+
+        block_for_each(mrModelPart.Elements(), [&](Element& r_element){
+            // If the values are set element-wise have priority
+            bool requires_unique_initial_state = false;
+            if (r_element.GetGeometry().Has(INITIAL_STRAIN_VECTOR)) {
+                noalias(aux_initial_strain) = (r_element.GetGeometry()).GetValue(INITIAL_STRAIN_VECTOR);
+                requires_unique_initial_state = true;
+            }
+            if (r_element.GetGeometry().Has(INITIAL_STRESS_VECTOR)) {
+                noalias(aux_initial_stress) = (r_element.GetGeometry()).GetValue(INITIAL_STRESS_VECTOR);
+                requires_unique_initial_state = true;
+            }
+            if (r_element.GetGeometry().Has(INITIAL_DEFORMATION_GRADIENT_MATRIX)) {
+                noalias(aux_initial_F) = (r_element.GetGeometry()).GetValue(INITIAL_DEFORMATION_GRADIENT_MATRIX);
+                requires_unique_initial_state = true;
+            }
+
+            std::vector<ConstitutiveLaw::Pointer> constitutive_law_vector;
+            r_element.CalculateOnIntegrationPoints(CONSTITUTIVE_LAW, constitutive_law_vector, mrModelPart.GetProcessInfo());
+
+            if (requires_unique_initial_state) {
+                InitialState::Pointer p_initial_state_custom = Kratos::make_intrusive<InitialState>(aux_initial_strain, aux_initial_stress, aux_initial_F);
+                for (IndexType point_number = 0; point_number < r_integration_points.size(); ++point_number) {
+                    constitutive_law_vector[point_number]->SetInitialState(p_initial_state_custom);
+                }
+            } else {
+                for (IndexType point_number = 0; point_number < r_integration_points.size(); ++point_number) {
+                    constitutive_law_vector[point_number]->SetInitialState(p_initial_state);
+                }
+            }
+        });
+
+        // #pragma omp parallel for
+        // for (int i = 0; i < static_cast<int>(mrModelPart.Elements().size()); i++) {
+        //     auto it_elem = it_elem_begin + i;
+
+        //     // If the values are set element-wise have priority
```

Comment:
are these commented-out  lines necessary?

Reply:
yes, to see If you are **checking** ;) I'll remove them, THX